presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkQueryExecutionFactory.java

```diff
@@ -684,6 +685,78 @@ private PrestoSparkQueryExecution(
             return result.build();
         }
 
+        @Override
+        public Stream<List<Object>> executeAsStream()
+        {
+            queryStateTimer.beginRunning();
+
+            List<Tuple2<MutablePartitionId, PrestoSparkSerializedPage>> rddResults;
+            try {
+                rddResults = doExecute(fragmentedPlan);
```

Comment:
This assumes that the results are already loaded inmemory, as it returns the `List` of tuples. I'm not even sure how well is iterative results fetching is supported in Spark.